rskj-core/src/main/java/org/ethereum/vm/program/Program.java

```diff
@@ -461,32 +478,74 @@ public void createContract(DataWord value, DataWord memStart, DataWord memSize)
         // Start tracking repository changes for the constructor of the contract
         Repository track = getStorage().startTracking();
 
+        ProgramResult programResult = ProgramResult.empty();
+
         //In case of hashing collisions, check for any balance before createAccount()
-        if (track.isExist(newAddress)) {
-            Coin oldBalance = track.getBalance(newAddress);
-            track.createAccount(newAddress);
-            track.addBalance(newAddress, oldBalance);
+        if (track.isExist(contractAddress)) {
+            // Hashing collisions in CREATE are rare, but in CREATE2 are possible
+            // We check for the nonce to be zero and that the code to be empty
+            // if any of this conditions fails, the contract creation fails:
+            // We set an exception as the program result and set the program
+            // to empty so it does not executes
+
+
+            byte[] code = track.getCode(contractAddress);
+
+            if (getActivations().isActive(ConsensusRule.RSKIP125) && ((code == null || code.length == 0) ||
```

Comment:
" and the destination address already has either nonzero nonce, or *nonempty code*, then the creation throws *immediately*"